src/python/py_vlm_pipeline.cpp

```diff
@@ -55,59 +55,73 @@ auto vlm_generate_kwargs_docstring = R"(
+            params.insert({ov::genai::image(std::move(py::cast<ov::Tensor>(value)))});
         } else if (key == "generation_config") {
-            params.insert({ov::genai::generation_config(std::move(py::cast<ov::genai::GenerationConfig>(item.second)))});
+            params.insert({ov::genai::generation_config(std::move(py::cast<ov::genai::GenerationConfig>(value)))});
         } else if (key == "streamer") {
             auto py_streamer = py::cast<utils::PyBindStreamerVariant>(value);
             params.insert({ov::genai::streamer(std::move(ov::genai::pybind::utils::pystreamer_to_streamer(py_streamer)))});
-
-        } else {
-            throw(std::invalid_argument("'" + key + "' is unexpected parameter name. "
+        } 
+        else {
+            if (allow_compile_properties) {
+                // convert arbitrary objects to ov::Any
+                // not supported properties are not checked, as these properties are passed to compile(), which will throw exception in case of unsupported property
+                if (utils::py_object_is_any_map(value)) {
+                    auto map = utils::py_object_to_any_map(value);
+                    params.insert(map.begin(), map.end());
+                } else {
+                    params[key] = utils::py_object_to_any(value);
+                }
+            }
+            else {
+                // generate doesn't run compile(), so only VLMPipeline specific properties are allowed
+                throw(std::invalid_argument("'" + key + "' is unexpected parameter name. "
                                         "Use help(openvino_genai.VLMPipeline.generate) to get list of acceptable parameters."));
+            }
         }
     }
+    if (config.size()) {
+        auto properties = utils::properties_to_any_map(config);
+        params.insert(properties.begin(), properties.end());
+    }
+    return params;
+}
+
+py::object call_vlm_generate(
+    ov::genai::VLMPipeline& pipe, 
+    const std::string& prompt,
+    const std::vector<ov::Tensor>& images,
+    const ov::genai::GenerationConfig& generation_config, 
+    const utils::PyBindStreamerVariant& py_streamer, 
+    const py::kwargs& kwargs
+) {
+    auto updated_config = *ov::genai::pybind::utils::update_config_from_kwargs(generation_config, kwargs);
+    ov::genai::StreamerVariant streamer = ov::genai::pybind::utils::pystreamer_to_streamer(py_streamer);
     
-    return py::cast(pipe.generate(prompt, params));
+    return py::cast(pipe.generate(prompt, images, updated_config, streamer));
 }
 
 void init_vlm_pipeline(py::module_& m) {
     py::class_<ov::genai::VLMPipeline>(m, "VLMPipeline", "This class is used for generation with VLMs")
         .def(py::init([](
             const std::string& model_path, 
             const std::string& device,
-            const std::map<std::string, py::object>& config
+            const std::map<std::string, py::object>& config,
```

Comment:
I think `kwargs` approach is more generic and we can drop `config`

as for `LLMPipeline` - let's try to "deprecate" `config` somehow to be aligned how pipelines are configured